pkg/client/tables.go

```diff
@@ -234,6 +236,73 @@ func (c *Client) NewTableEntry(
 	return entry
 }
 
+func (c *Client) ReadTableEntry(table string, mfs []MatchInterface) (*p4_v1.TableEntry, error) {
+	tableID := c.tableId(table)
+
+	entry := &p4_v1.TableEntry{
+		TableId: tableID,
+	}
+
+	for idx, mf := range mfs {
+		entry.Match = append(entry.Match, mf.get(uint32(idx+1), c.CanonicalBytestrings))
+	}
+
+	entity := &p4_v1.Entity{
+		Entity: &p4_v1.Entity_TableEntry{TableEntry: entry},
+	}
+
+	readEntity, err := c.ReadEntitySingle(entity)
+	if err != nil {
+		return nil, fmt.Errorf("error when reading table entry: %v", err)
+	}
+
+	readEntry := readEntity.GetTableEntry()
+	if readEntry == nil {
+		return nil, fmt.Errorf("server returned an entity but it is not a table entry! ")
+	}
+
+	return readEntry, nil
+}
+
+func (c *Client) ReadTableEntryWildcard(table string) ([]*p4_v1.TableEntry, error) {
+	tableID := c.tableId(table)
+
+	entry := &p4_v1.TableEntry{
+		TableId: tableID,
+	}
+
+	out := make([]*p4_v1.TableEntry, 0)
+	readEntityCh := make(chan *p4_v1.Entity, 100)
+
+	var wg sync.WaitGroup
+	var err error
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+		for readEntity := range readEntityCh {
+			readEntry := readEntity.GetTableEntry()
+			if readEntry == nil {
+				err = fmt.Errorf("server returned an entity which is not a table entry! ")
+				break
+			}
```

Comment:
If you stop reading from the channel here, there is a possibility that `ReadEntityWildcard` will keep blocking and not return? I realize that there is the same issue in the `ReadEntityWildcard` function which I wrote... so I can fix both in a follow-up PR